server/src/main/java/org/apache/druid/client/ImmutableDruidDataSource.java

```diff
@@ -120,8 +120,39 @@ public String toString()
            + "'}";
   }
 
+  /**
+   *
+   * ImmutableDruidDataSource should be considered a container, not a data class.
+   * The idea is the same as behind prohibiting/limiting equals() (and therefore usage as HashSet/HashMap keys) of DataSegment: see #6358.
+   * When somebody wants to deduplicate ImmutableDruidDataSource objects, they would need to put them into a Map<String, ImmutableDruidDataSource> and resolve conflicts by name manually.
```

Comment:
Lines should be shorter 120 cols. Please configure your IDE and check all code added in this PR.

Reply:
Fixed.